sklearn/utils/estimator_checks.py

```diff
@@ -324,6 +324,10 @@ def set_checking_parameters(estimator):
         # SVC
         estimator.set_params(decision_function_shape='ovo')
 
+    if "include_self" in params:
+        # KNeighborsTransformer
+        estimator.set_params(include_self=True)
```

Comment:
What does this fix?

Reply:
This fixes the common transformer test on fit_transform and fit+transform:
```py
trans = KNeighborsTransformer(include_self=True) 
trans.fit(X).transform(X) == trans.fit_transform(X)  # True

trans = KNeighborsTransformer(include_self=False)  # default
trans.fit(X).transform(X) == trans.fit_transform(X)  # False
```

As I don't see how it could be possible to comply with this test, I change the default parameter in the test.

Reply:
I thought I replied to this: we are forced to break this invariance elsewhere, including KNNImputer (#9212) for the same reasons, and stacking transformer (#8960) because `fit_transform` needs to use blended predictions (i.e. `cross_val_predict`) while `transform` uses a single model fit on the entire training data. So we instead need an exceptions list (or estimator tag) here.